### PR TITLE
googleads/google-ads-php#98 Added more resource names

### PR DESCRIPTION
**Pull request by MaksTech**

### CODE REVIEW

File: src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php [hunk 1]
```diff
@@ -17,17 +17,57 @@
...
 
+use Google\Ads\GoogleAds\V1\Services\AccountBudgetServiceClient;
+use Google\Ads\GoogleAds\V1\Services\AccountBudgetProposalServiceClient;
+use Google\Ads\GoogleAds\V1\Services\AdGroupServiceClient;
 use Google\Ads\GoogleAds\V1\Services\AdGroupAdServiceClient;
+use Google\Ads\GoogleAds\V1\Services\AdGroupAdLabelServiceClient;
+use Google\Ads\GoogleAds\V1\Services\AdGroupAudienceViewServiceClient;
 use Google\Ads\GoogleAds\V1\Services\AdGroupBidModifierServiceClient;
 use Google\Ads\GoogleAds\V1\Services\AdGroupCriterionServiceClient;
-use Google\Ads\GoogleAds\V1\Services\AdGroupServiceClient;
```
Comment (Raibaz) [inline src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: Please preserve the alphabetic sorting of uses.

File: src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php [hunk 2]
```diff
@@ -60,9 +124,55 @@ public static function forAdGroup($customerId, $adGroupId)
      * @param int $adId the ad ID
      * @return string the ad group ad resource name
      */
-    public static function forAdGroupAd($customerId, $adGroupId, $adId)
-    {
-        return AdGroupAdServiceClient::adGroupAdName($customerId, "{$adGroupId}~{$adId}");
+    public static function forAdGroupAd(
```
Comment (Raibaz) [inline src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: These can fit on a single line.

File: src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php [hunk 3]
```diff
@@ -89,14 +202,218 @@ public static function forAdGroupBidModifier($customerId, $adGroupId, $criterion
      * @param int $criterionId the criterion ID
      * @return string the ad group criterion resource name
      */
-    public static function forAdGroupCriterion($customerId, $adGroupId, $criterionId)
-    {
+    public static function forAdGroupCriterion(
```
Comment (Raibaz) [inline src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: These can fit on a single line.

File: src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php [hunk 4]
```diff
@@ -89,14 +202,218 @@ public static function forAdGroupBidModifier($customerId, $adGroupId, $criterion
...
+
+    /**
+     * Generates resource name for an ad group feed.
+     *
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID
+     * @param int $feedId the feed Id
+     * @return string the ad group feed resource name
+     */
+    public static function forAdGroupFeed(
```
Comment (Raibaz) [inline src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: These can fit on a single line.

**Comment by MaksTech** [comment]
Are methods also supposed to be in the same order as the use clauses? So in alphabetic order too

**Comment by Raibaz** [comment]
Yep having the methods in alphabetic order is helpful to find them quickly, thanks!

**Comment by MaksTech** [comment]
Okay, I'm also going to add the rest of the resource names

**Comment by MaksTech** [comment]
Just noticed, [PaymentsAccount](https://github.com/googleads/google-ads-php/blob/master/src/Google/Ads/GoogleAds/V1/Services/Gapic/PaymentsAccountServiceGapicClient.php) doesn't have template, name and parse name methods. Should I add them?

**Comment by Raibaz** [comment]
No, please don't add them: the `PaymentsAccount` class is actually generated by the proto compiler, so any manual change would be lost in the next release. 

Besides, it's intentional for it not to have the template, name and parse methods as the PaymentsAccount resource cannot be queried by itself but only as part of a billing setup.

File: src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php [hunk 5]
```diff
@@ -162,88 +504,1006 @@ public static function forCampaignCriterion($customerId, $campaignId, $criterion
...
+    public static function forParentalStatusView($customerId, $adGroupId, $criterionId)
+    {
+        return ParentalStatusViewServiceClient::parentalStatusViewName(
+            $customerId,
+            "{$adGroupId}~{$criterionId}"
+        );
+    }
+
+    /**
+     * Generates resource name for a paid organic search term view.
```
Comment (fiboknacky) [inline src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: Please fix wrong PHPDoc here and in @return

Comment (MaksTech) [reply src/Google/Ads/GoogleAds/Util/V1/ResourceNames.php]: Done

**Pull request merged**